c-dependencies/js-compute-runtime/builtins/backend.cpp

```diff
@@ -17,6 +19,617 @@
 #include "host_call.h"
 #include "js/Conversions.h"
 
+enum class Authentication {
+  RSA,
+};
+
+enum class KeyExchange {
+  EECDH,
+  RSA,
+};
+
+enum class Encryption {
+  AES128,
+  AES128GCM,
+  AES256,
+  AES256GCM,
+  CHACHA20POLY1305,
+  TRIPLE_DES,
+};
+
+enum class EncryptionLevel {
+  MEDIUM,
+  HIGH,
+};
+
+enum class MessageDigest {
+  SHA1,
+  SHA256,
+  SHA384,
+  AEAD,
+};
+
+enum class Protocol {
+  SSLv3,
+  TLSv1,
+  TLSv1_2,
+};
+
+class Cipher {
+  std::string openSSLAlias;
+  KeyExchange kx;
+  Authentication au;
+  Encryption enc;
+  MessageDigest mac;
+  Protocol protocol;
+  EncryptionLevel level;
+  int strength_bits;
+
+public:
+  Cipher(std::string openSSLAlias, KeyExchange kx, Authentication au, Encryption enc,
+         MessageDigest mac, Protocol protocol, EncryptionLevel level, int strength_bits)
+      : openSSLAlias(openSSLAlias), kx(kx), au(au), enc(enc), mac(mac), protocol(protocol),
+        level(level), strength_bits(strength_bits) {}
+
+  const std::string &getOpenSSLAlias() { return openSSLAlias; }
+
+  KeyExchange getKx() { return kx; }
+
+  Authentication getAu() { return au; }
+
+  Encryption getEnc() { return enc; }
+
+  MessageDigest getMac() { return mac; }
+
+  Protocol getProtocol() { return protocol; }
+
+  EncryptionLevel getLevel() { return level; }
+
+  int getStrength_bits() { return strength_bits; }
+
+  // Overload the == operator
+  const bool operator==(const Cipher &obj) const {
+    return openSSLAlias == obj.openSSLAlias && kx == obj.kx && au == obj.au && enc == obj.enc &&
+           mac == obj.mac && protocol == obj.protocol && level == obj.level &&
+           strength_bits == obj.strength_bits;
+  }
+};
+
+/**
+ * Class in charge with parsing openSSL expressions to define a list of ciphers.
+ */
+class OpenSSLCipherConfigurationParser {
+private:
+  std::unordered_map<std::string, std::vector<Cipher>> aliases;
```

Comment:
This type gets used all over the place, and would benefit from having an alias introduced.

```suggestion
  using AliasMap = std::unordered_map<std::string, std::vector<Cipher>>;
  AliasMap aliases;
```

Reply:
Do you have a sense for how big this map gets? It might be just as efficient to represent it as a vector of pairs, instead of using an unordered map.

Reply:
It currently is a fixed size of 49 entries, which will change whenever a cipher is added or removed from Fastly's supported cipher list